### DISCUSSION TITLE
Challenges in Wrapping AccordionTab within Custom Accordion Component

### DISCUSSION THREAD

**Discussion by james-cavanagh** [primefaces/community#726] [PrimeVue]
I am currently developing a custom wrapper component for the Accordion and AccordionTab components. The goal is to enhance these components with additional styling and functionality. However, I've encountered an issue related to the internal workings of the Accordion component.

The Accordion component uses a method `isAccordionTab(child)` to verify if a child component is an AccordionTab. This method checks the child's type name:

```js
isAccordionTab(child) {
    return child.type.name === 'AccordionTab';
}
```

When wrapping the AccordionTab component in my custom component, this check fails because the type name of the child changes, and as a result, the Accordion no longer recognises it as an AccordionTab.

I am looking for guidance or suggestions on how to properly wrap the AccordionTab component in a way that the Accordion can still recognise it as an AccordionTab. Are there recommended practices or potential updates to PrimeVue that could facilitate this kind of component wrapping without losing the inherent functionalities?

Additionally, I have noticed a similar issue when attempting to wrap other components, namely the Splitter, SplitterPanel, and TabView. These components also seem to have internal logic dependent on specific component names or types, which leads to complications when creating custom wrapper components for them. This suggests a broader pattern in the component library where wrapping components for customisation can disrupt the expected functionality.

**Comment by fhnaumann** [comment]
The problem is also discussed here: https://github.com/primefaces/primevue/issues/4646
For some reason it was implemented but then reverted...

**Reply by james-cavanagh** [reply]
Thanks, I wasn't aware of that issue and couldn't find anything. I will close this off.

**Comment by james-cavanagh** [comment]
Closing as this is an active issue - https://github.com/primefaces/primevue/issues/4646
